File: Python/ExportAndImportSample/ExportAndImport/MetadataMethods.py
#! /usr/bin/env python3
# pylint: disable= no-name-in-module

import json

from restapisdk.models.export_object_tml_export_associated_enum import ExportObjectTMLExportAssociatedEnum
from restapisdk.models.import_object_tml_import_policy_enum import ImportObjectTMLImportPolicyEnum
from restapisdk.models.name_and_id_input import NameAndIdInput
from restapisdk.models.search_object_header_type_enum import SearchObjectHeaderTypeEnum
from restapisdk.models.tspublic_rest_v_2_metadata_header_search_request \
    import TspublicRestV2MetadataHeaderSearchRequest
from restapisdk.models.tspublic_rest_v_2_metadata_tml_export_request \
    import TspublicRestV2MetadataTmlExportRequest
from restapisdk.models.tspublic_rest_v_2_metadata_tml_import_request \
    import TspublicRestV2MetadataTmlImportRequest

from Python.ExportAndImportSample.ExportAndImport.ControllerBase \
    import ControllerBase


class MetadataMethods(ControllerBase):

    def SearchObject(self, client, p_Name):
        metadataController = client.metadata
        body = TspublicRestV2MetadataHeaderSearchRequest()
        body.output_fields = ['id']
        body.offset = 3
        body.created_by = []
        body.created_by.append(NameAndIdInput())
        body.created_by[0].name = p_Name
        body.mtype = SearchObjectHeaderTypeEnum.LIVEBOARD
        searchIds = []
        result = metadataController.search_object_header(body)
        data_dict = json.loads(result)
        jsonArray = data_dict['headers']
        for data in jsonArray:
            searchIds.append(data['id'])
        return searchIds

    def ExportObject(self, client, searchIds):
        metadataController = client.metadata

        body = TspublicRestV2MetadataTmlExportRequest()
        body.id = searchIds
        body.export_associated = ExportObjectTMLExportAssociatedEnum.TRUE

        result = metadataController.export_object_tml(body)
        return result

    def ImportObject(self, client, p_Tmls):

        metadataController = client.metadata
        body = TspublicRestV2MetadataTmlImportRequest()
        body.import_policy = ImportObjectTMLImportPolicyEnum.VALIDATE_ONLY
        body.object_tml = p_Tmls

        result = metadataController.import_object_tml(body)
        return result

File: DotNet/ExportAndImportSample/MetadataMethods.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Globalization;
using System.Threading.Tasks;
using System.Collections;
using System.Reflection;
using Newtonsoft.Json.Linq;
using System.Linq;

using RESTAPISDK.Standard;
using RESTAPISDK.Standard.Models;
using RESTAPISDK.Standard.Controllers;
using RESTAPISDK.Standard.Utilities;
using RESTAPISDK.Standard.Exceptions;


namespace ExportAndImport
{
    class MetadataMethods : ControllerBase
    {

        public async Task<List<string>> SearchObject(RESTAPISDKClient client, string p_name)
        {
            MetadataController metadataController = client.MetadataController;

            var body = new TspublicRestV2MetadataHeaderSearchRequest();
            //Add type of object
            body.Type = SearchObjectHeaderTypeEnum.LIVEBOARD;

            //Include only id in the output
            body.OutputFields = new List<string>();
            body.OutputFields.Add("id");

            //Set offset and batch size
            body.Offset = 0;
            body.BatchSize = 5;

            //Add the created by name
            body.Author = new List<NameAndIdInput>();
            var bodyCreatedBy = new NameAndIdInput();
            bodyCreatedBy.Name = p_name;
            body.Author.Add(bodyCreatedBy);

            //Send search request based on the inputs above
            object result = await metadataController.SearchObjectHeaderAsync(body);

            //Parse the response to get id and name
            var resultJObject = JObject.Parse(result.ToString());
            List<string> l_id = new List<string>();

            if (resultJObject != null)
            {
                var objectArray = resultJObject["headers"];

                foreach (object item in objectArray)
                {
                    var idJObject = JObject.Parse(item.ToString());
                    var objectId = idJObject["id"];
                    l_id.Add(objectId.ToString());
                }
            }

            return l_id;
        }

        public async Task<object> ExportObject(RESTAPISDKClient p_client, List<string> p_id)
        {

            MetadataController metadataController = p_client.MetadataController;

            var body = new TspublicRestV2MetadataTmlExportRequest();
            body.Id = p_id;
            body.ExportAssociated = ExportObjectTMLExportAssociatedEnum.True;

            object result = await metadataController.ExportObjectTMLAsync(body);

            return result;
        }

        public async Task<string> ImportObject(RESTAPISDKClient p_client, List<string> p_tmls)
        {

            MetadataController metadataController = p_client.MetadataController;

            var body = new TspublicRestV2MetadataTmlImportRequest();
            body.ObjectTML = p_tmls;
            body.ImportPolicy = ImportObjectTMLImportPolicyEnum.VALIDATEONLY;

            object result = await metadataController.ImportObjectTMLAsync(body);

            return result.ToString();
        }
    }
}
